lnwallet/chainfee/minrelayfeemanager_test.go

```diff
@@ -0,0 +1,54 @@
+package chainfee
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+type mockChainBackend struct {
+	minRelayFee SatPerKWeight
+	callCount   int
+}
+
+func (m *mockChainBackend) fetchFee() (SatPerKWeight, error) {
+	m.callCount++
+	return m.minRelayFee, nil
+}
+
+// TestMinRelayFeeManager tests that the minRelayFeeManager returns an up to
+// date min relay fee by querying the chain backend and that it returns a cached
+// fee if the chain backend was recently queried.
+func TestMinRelayFeeManager(t *testing.T) {
+	chainBackend := &mockChainBackend{
+		minRelayFee: SatPerKWeight(1000),
+	}
+
+	feeManager := &minRelayFeeManager{
+		minUpdateInterval: 100 * time.Millisecond,
+		fetchNewFee:       chainBackend.fetchFee,
+	}
+
+	// Init the min relay fee manager. This should call the chain backend
+	// once.
+	err := feeManager.initMinRelayFee()
+	require.NoError(t, err)
+	require.Equal(t, 1, chainBackend.callCount)
+
+	// If the fee is requested again, the stored fee should be returned
+	// and the chain backend should not be queried.
+	chainBackend.minRelayFee = SatPerKWeight(2000)
+	minRelayFee := feeManager.getFee()
+	require.Equal(t, minRelayFee, SatPerKWeight(1000))
+	require.Equal(t, 1, chainBackend.callCount)
+
+	time.Sleep(200 * time.Millisecond)
```

Comment:
One way to get rid of the sleep here would just be updating the last update time in place.